### PR TITLE
Remove unused account analytics

### PR DESCRIPTION
**Pull request by eessex**
REVIEW APP: https://account-analytics.artsy.net/

Part of a larger effort to deprecate analytics events in `/desktop/analytics` and `/mobile/analytics`, these changes remove legacy analytics events related to account events. These events are all duplicated in React code, and no longer need to fire.

Also,
- converts `/mobile/assets/mobile_analytics` to typescript
- use additional helpers in/between desktop and mobile analytics

https://artsyproduct.atlassian.net/browse/PLATFORM-2689

### CODE REVIEW

File: src/desktop/apps/consign/client/analytics_middleware.ts [hunk 1]
```diff
@@ -1,24 +1,13 @@
...
-
-// FIXME: Rewire
-let analyticsHooks = _analyticsHooks
-
 const analyticsMiddleware = store => next => action => {
   const result = next(action)
   const nextState = store.getState()
 
   // track certain types of actions
   switch (action.type) {
```
Comment (eessex) [inline src/desktop/apps/consign/client/analytics_middleware.ts:9]: This action may be safe to deprecate.

**Comment by abhitip** [comment]
Just checked the fair page for those two events that were still firing and they are no longer firing so I think we are good!

File: src/desktop/analytics/helpers.ts [hunk 2]
```diff
@@ -35,6 +37,32 @@ export const trackEvent = (data: any, options: object = {}) => {
...
+    })
+    // clear analytics cache when user logs out
+    analyticsHooks.on("auth:logged-out", () => window.analytics.reset())
+  }
+}
+
+/**
+ * Send a pageview instead of track event when a stub is expanded,
+ * for example clicking read more in articles infinite scroll
+ */
```
Comment (damassi) [inline src/desktop/analytics/helpers.ts:65]: 👍 ty for the comment

File: src/desktop/apps/consign/client/analytics_middleware.ts [hunk 3]
```diff
@@ -1,24 +1,13 @@
...
     case actions.UPDATE_USER: {
-      const { user, accountCreated } = action.payload
-      analyticsHooks.trigger("consignment:account:created", {
-        id: user.id,
-        email: user.email,
-        accountCreated: accountCreated,
-      })
```
Comment (damassi) [inline src/desktop/apps/consign/client/analytics_middleware.ts:21]: We're no longer tracking consignment account creation?

File: src/desktop/apps/consign/client/analytics_middleware.ts (hunk 1 above)
Comment (damassi) [reply src/desktop/apps/consign/client/analytics_middleware.ts:9]: @eessex - Mind elaborating a bit on the "may"? 😄

File: src/desktop/apps/consign/client/analytics_middleware.ts (hunk 3 above)
Comment (eessex) [reply src/desktop/apps/consign/client/analytics_middleware.ts:21]: This is a duplicate. Consign used to have a custom auth implementation, and this event was not removed when we updated the flow to use the global auth component.

File: src/desktop/apps/consign/client/analytics_middleware.ts (hunk 1 above)
Comment (eessex) [reply src/desktop/apps/consign/client/analytics_middleware.ts:9]: i didn't want to touch the consign code too much, but i'm assuming that firing the analytics event is the only reason this action exists.

File: src/desktop/apps/consign/client/analytics_middleware.ts (hunk 3 above)
Comment (damassi) [reply src/desktop/apps/consign/client/analytics_middleware.ts:21]: I see, so it's tracked in a separate component 👍

File: src/desktop/apps/consign/client/analytics_middleware.ts (hunk 1 above)
Comment (damassi) [reply src/desktop/apps/consign/client/analytics_middleware.ts:9]: We have a few actions we're tracking -- SUBMISSION_ERROR, SUBMISSION_CREATED , SUBMISSION_COMPLETED, etc. Do you mean all of these can be deprecated?

Comment (eessex) [reply src/desktop/apps/consign/client/analytics_middleware.ts:9]: Sorry placement of comment was unclear, `UPDATE_USER` is the action that I believe can be deprecated.

File: src/desktop/assets/analytics.ts [hunk 4]
```diff
@@ -42,26 +41,7 @@ require("../analytics/main_layout.ts")
...
-      })
-      window.analytics.identify(sd.CURRENT_USER.id, traits, {
-        integrations: { Marketo: false },
-      })
-      // clear analytics cache when user logs out
-      analyticsHooks.on("auth:logged-out", () => window.analytics.reset())
-    }
-
+    onAnalyticsReady()
     setupSplitTests()
```
Comment (eessex) [inline src/desktop/assets/analytics.ts:46]: Will investigate this in a separate PR but wanted to note that this `setupSplitTests` is not present in mobile-- it is possible that split tests have been skipping pages entirely if they use the legacy mobile templates.

**Pull request merged**